lib/rubocop/config.rb

```diff
@@ -405,5 +401,28 @@ def check_target_ruby
 
       raise ValidationError, msg
     end
+
+    def reject_mutually_exclusive_defaults
+      disabled_by_default = for_all_cops['DisabledByDefault']
+      enabled_by_default = for_all_cops['EnabledByDefault']
+      return unless disabled_by_default && enabled_by_default
+
+      msg = 'Cops cannot be both enabled by default and disabled by default'
+      raise ValidationError, msg
+    end
+
+    def enable_cop?(qualified_cop_name, cop_options)
+      cop_department, cop_name = qualified_cop_name.split('/')
+      department = cop_name.nil?
+
+      unless department
+        department_options = self[cop_department]
+        if department_options && false == department_options['Enabled']
```

Comment:
Yoda conditions protect against using `=` when you mean `==`.  I was under the impression that they are recommended.

Is this documented in the styleguide?